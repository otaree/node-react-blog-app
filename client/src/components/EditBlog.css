.EditBlog {
    width: 80%;
    margin: 0 auto;
}

.EditBlog h2 {
    text-align: center;
}


.delete-blog {
    margin-left: 41.5%;
}

.btn-delete {
    outline: none;
    border: 1px solid #888;
    border-radius: 3px;
    background-color: rgb(189, 48, 48);
    font-size: 16px;
    font-weight: bold;
    padding: 10px;
    width: 120px;
    color: #eee;

}

.btn-delete:hover {
    background-color: lightcoral;
    color: #444;
    cursor: pointer;
}