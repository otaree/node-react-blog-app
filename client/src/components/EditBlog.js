import React from 'react';
import { connect } from 'react-redux';
import Form from './Form';
import * as actions from '../actions/blog'; 
import "./EditBlog.css";

export class EditBlog extends React.Component {
    onSubmit = (updateBlogData) => {
        this.props.updateBlog(this.props.id, updateBlogData);
        this.props.history.push('/');
    }

    onClick = () => {
        this.props.deleteBlog(this.props.id);
        this.props.history.push('/');
    }
    render() {
        return (
            <section className="EditBlog">
                    <h2>Edit Blog</h2>
                    <Form blog={this.props.blog} onSubmit={this.onSubmit} />
                    <div className="delete-blog">
                        <button
                            className="btn-delete"
                            onClick={this.onClick}  
                        >
                            Delete Blog
                        </button>
                    </div>
            </section>
        );
    }
}

const mapStateToProps = (state, wrapperProps) => {
    const id = wrapperProps.match.params.id;
    return {
        blog: state.blogs.find(blog => blog._id === id ),
        id
    };
};

const mapDispatchToProps = dispatch => {
    return {
        updateBlog: (id, blogData) => dispatch(actions.patchBlog(id, blogData)),
        deleteBlog: id => dispatch(actions.deleteBlog(id)) 
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(EditBlog);