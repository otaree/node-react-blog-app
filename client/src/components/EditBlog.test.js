import React from 'react';
import { shallow } from 'enzyme';
import { EditBlog } from './EditBlog';

const blogs = [
    {
        _id: '1',
        title: "First Title",
        body: "First body",
        author: "author1"
    },
    {
        _id: '2',
        title: "Second Title",
        body: "Second body",
        author: "author2"
    }
];

let editBlog, props, mockUpdateBlog, mockDeleteBlog, mockPush;

describe('EditBlog', () => {
    beforeEach(() => {
        mockUpdateBlog = jest.fn()        
        mockDeleteBlog = jest.fn();
        mockPush = jest.fn();
        props = {
            updateBlog: mockUpdateBlog,
            deleteBlog: mockDeleteBlog,
            history: {
                push: mockPush
            },
            blog: blogs[0],
            id: blogs[0]._id
        }
        editBlog = shallow(<EditBlog {...props} />);
        
    });

    it('renders properly', () => {
        expect(editBlog).toMatchSnapshot();
    });

    it('should handle `updateBlog`', () => {
        const updateData = {
            title: "Test Update",
            body: "Test body",
            author: "testAuthor",
            published: false
        };
        editBlog.find("Form").prop('onSubmit')(updateData);
        expect(mockPush).toHaveBeenLastCalledWith('/');
        expect(mockUpdateBlog).toHaveBeenLastCalledWith(blogs[0]._id, updateData);
    });

    it('should handle `deleteBlog`', () => {
        editBlog.find('.btn-delete').simulate('click');
        expect(mockPush).toHaveBeenLastCalledWith('/');
        expect(mockDeleteBlog).toHaveBeenLastCalledWith(blogs[0]._id);
    });
});