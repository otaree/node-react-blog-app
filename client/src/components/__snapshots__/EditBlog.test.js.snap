// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditBlog renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditBlog
    blog={
        Object {
            "_id": "1",
            "author": "author1",
            "body": "First body",
            "title": "First Title",
          }
    }
    deleteBlog={[Function]}
    history={
        Object {
            "push": [Function],
          }
    }
    id="1"
    updateBlog={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Edit Blog
</h2>,
        <Form
          blog={
                    Object {
                              "_id": "1",
                              "author": "author1",
                              "body": "First body",
                              "title": "First Title",
                            }
          }
          onSubmit={[Function]}
/>,
        <div
          className="delete-blog"
>
          <button
                    className="btn-delete"
                    onClick={[Function]}
          >
                    Delete Blog
          </button>
</div>,
      ],
      "className": "EditBlog",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Edit Blog",
        },
        "ref": null,
        "rendered": "Edit Blog",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "blog": Object {
            "_id": "1",
            "author": "author1",
            "body": "First body",
            "title": "First Title",
          },
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="btn-delete"
            onClick={[Function]}
>
            Delete Blog
</button>,
          "className": "delete-blog",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Delete Blog",
            "className": "btn-delete",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Delete Blog",
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Edit Blog
</h2>,
          <Form
            blog={
                        Object {
                                    "_id": "1",
                                    "author": "author1",
                                    "body": "First body",
                                    "title": "First Title",
                                  }
            }
            onSubmit={[Function]}
/>,
          <div
            className="delete-blog"
>
            <button
                        className="btn-delete"
                        onClick={[Function]}
            >
                        Delete Blog
            </button>
</div>,
        ],
        "className": "EditBlog",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Edit Blog",
          },
          "ref": null,
          "rendered": "Edit Blog",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "blog": Object {
              "_id": "1",
              "author": "author1",
              "body": "First body",
              "title": "First Title",
            },
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn-delete"
              onClick={[Function]}
>
              Delete Blog
</button>,
            "className": "delete-blog",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Delete Blog",
              "className": "btn-delete",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Delete Blog",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
